app/js/addCategories.ts

```diff
@@ -0,0 +1,45 @@
+async function addCategory ():Promise<void> {
+    let categoriesObject = await getData('/category');
+    document.querySelector<HTMLFormElement>('.categoriesForm').addEventListener('submit', (e) => {
+        e.preventDefault();
+        let newCategory: string = document.querySelector<HTMLFormElement>('#categoryName').value;
+        let responseMessage = document.querySelector('#addedCategoryConfirmation');
```

Comment:
this doesnt need to be done inside the event listener. the element never changes so do it in the function before the listener